docsk8s/quick-start/ls-k8s-quick-start.asciidoc

```diff
@@ -0,0 +1,203 @@
+--
+
+Your `logstash-k8s-gs/cert` folder should now contain a set of certificate files, including `client` certificates for Filebeat and Metricbeat, and `server` certificates for Logstash. 
+
+The parent `logstash-k8s-gs` directory also has a new `001-secret.yaml` resources file that stores a hash of the client and server certificates.
+
+image::./images/gs-cert-files.png[generated CA certificate files]
+
+[float]
+[[qs-create-elastic-stack]]
+=== Create an Elastic Stack
+
+Now that your environment and certificates are set up, it's time to create an Elastic Stack. Run the following command to deploy the example using the sample CRDs:
+
+[source,sh]
+--
+kubectl apply -f .
+--
+
+The resources are created:
+
+[source,sh]
+--
+elasticsearch.elasticsearch.k8s.elastic.co/demo created
+configmap/logstash-pipeline created
+configmap/logstash-config created
+secret/logstash-beats-tls created
+deployment.apps/logstash created
+service/logstash created
+horizontalpodautoscaler.autoscaling/logstash created
+beat.beat.k8s.elastic.co/demo created
+beat.beat.k8s.elastic.co/demo configured
+kibana.kibana.k8s.elastic.co/demo created
+--
+
+[source,sh]
+--
+kubectl get pods
+--
+
+The pods are starting up. You may need to wait a minute or two for all of them to be ready.
+
+[source,sh]
+--
+NAME                                    READY   STATUS    RESTARTS       AGE
+demo-beat-filebeat-7f4d97f69f-qkkbl     1/1     Running   0              42s
+demo-beat-metricbeat-59f4b68cc7-9zrrn   1/1     Running   0              39s
+demo-es-default-0                       1/1     Running   0              41s
+demo-kb-d7f585494-vbf6s                 1/1     Running   0              39s
+elastic-operator-0                      1/1     Running   4 (164m ago)   13d
+logstash-7974b9ccb9-jd5xl               1/1     Running   0              42s
+--
+
+[float]
+[[qs-view-monitoring-data]]
+=== View the stack monitoring data
+
+Now that your stack monitoring data is flowing, let's access it in {kib}. 
+
+First, enable port forwarding for the {kib} service on port `5601`. Open a second shell window and run the following:
+
+[source,sh]
+--
+kubectl port-forward service/demo-kb-http 5601
+--
+
+Then, open up a web browser at address `https://localhost:5601`. Depending on your browser you may need to accept the site certificate.
+
+Log in to {kib} using the `elastic` username and password. To obtain the password, run:
+
+[source,sh]
+--
+kubectl get secret demo-es-elastic-user -o=jsonpath='{.data.elastic}' | base64 --decode; echo
+--
+
+Open the {kib} main menu and select **Management**, then **Stack Monitoring**.
+
+Select the {ls} **Overview**, and under the **Nodes** tab select the link for the {ls} node.
+
+image::./images/gs-logstash-node-metrics.png[{ls} metrics data in {kib}]
+
+That's it! The Kubernetes API server metrics data is flowing through {ls} into {es} and {kib}. You can monitor the JVM Heap, CPU Utilization, and System Load data as it updates in real time.
+
```

Comment:
We can add a session to explain how the config of each component works together. Reference: https://github.com/elastic/logstash/tree/feature/kubernetes/k8s/example/beats_ls_es#configuration

I can work on this part if it is too many technical details. WDYT?

Reply:
My thoughts here were to go through the config files, applying the main stack config together, then going through the ones that are relevant to logstash, one at a time, annotating the interesting parts

Something along the lines of:

```
Firstly, we'll create the `ConfigMaps` to hold logstash settings and pipeline configuration that we will refer to in the deployment configuration file. 

NOTE: we recommend setting custom JVM settings in an `LS_JAVA_OPTS` environment variable defined in the deployment configuration file, rather than creating a separate `ConfigMap` for `jvm.options`

`001-configmap.yaml`::
This contains the Logstash settings and pipeline configuration:
+
[source,yaml]
--
  ---
# ConfigMap for logstash pipeline definition
apiVersion: v1
kind: ConfigMap
metadata:
  name: logstash-pipeline
  labels:
    app: logstash-demo
data:
  logstash.conf: | <1>
    input {
      beats {
        port => "5044"
        ssl => true
        ssl_certificate_authorities => ["/usr/share/logstash/config/ca.crt"] <2> 
        ssl_certificate => "/usr/share/logstash/config/server.crt" <2>
        ssl_key => "/usr/share/logstash/config/server.pkcs8.key" <2>
        ssl_verify_mode => "force_peer"
      }
    }
    output {
      elasticsearch {
        hosts => ["https://demo-es-http:9200"]
        index => "kube-apiserver-%{+YYYY.MM.dd}"
        cacert => "/usr/share/logstash/config/es_ca.crt" <2>
        user => 'elastic'
        password => '${ELASTICSEARCH_PASSWORD}' <3>
      }
    }
---
# logstash.yml definition
apiVersion: v1
kind: ConfigMap
metadata:
  name: logstash-config
  labels:
    app: logstash-demo
data:
  logstash.yml: | <4>
    api.http.host: "0.0.0.0"

--

<1> This `data` section represents a full 
<2> Keys and certificates to secure communication between logstash and the rest of the stack. These are volumes that will be defined in the deployment configuration file
<3> The `ELASTICSEARCH_PASSWORD` is an environmental variable that will be defined in the deployment configuration file, using secrets created by the `Elasticsearch` CRD
<4> The `logstash.yml` `ConfigMap` should include all logstash.yml<LINK_TO_MAIN_DOC> settings that you wish to change from the default settings
```

WDYT?

(I may remove the boilerplate `apiVersion, kind, metadata` sections, and just stick to the `data:` section)

Reply:
Going through the Logstash files step by step sounds good to me!

Note that I stuck some detail about each of the config files, and how they're supposed to work together, in the [list of sample files](https://github.com/elastic/logstash/pull/14532/files#diff-44011a051f96a1797c66cf219b2cb83c8ee18dadc4e6500ff7716c88e27c9e1b). We may want to remove some of that content, or maybe just scrap that page altogether, if we're explaining most of the files on the main Quick Start page. Just a thought.